engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -541,9 +574,24 @@ private Class<?> calculateMethodReturnTypeUsingGenerics(
 
         final TypeVariable<?> genericReturnType = (TypeVariable<?>) methodReturnType;
 
+        // if the method is being invoked on a variable (i.e. NameExpr) of a parameterized type, and we have
+        // type arguments for that variable, then check whether the return type is one of the scope's type parameters.
+        if (scopeExpr instanceof NameExpr) {
+            Class<?>[] typeArguments = variableParameterizedTypes.get(((NameExpr) scopeExpr).getNameAsString());
+            if (typeArguments != null) {
+                final TypeVariable<? extends Class<?>>[] scopeTypeParameters = scope.getTypeParameters();
+                for (int i = 0; i < scopeTypeParameters.length; i++) {
+                    if (scopeTypeParameters[i].equals(genericReturnType)) {
+                        return typeArguments[i];
+                    }
+                }
+            }
+        }
```

Comment:
We should, but handling generics is a separate problem. (I have a DHE branch for that as well, but don't want to tie it into this PR, since it is also a big change.)

Reply:
you were right; I had to improve this now — it was a regression and caused some test failures